### ISSUE TITLE
mlReloadSchemas task deletes all the final DB content (DHF 4.1.0)

### ISSUE THREAD

**Issue by maachou** [marklogic/marklogic-data-hub#1716]
### The issue:
Calling the task mlReloadSchemas deletes all the final database content.

### The command executed:

  `gradle mlReloadSchemas`

### The command output:

gradle mlReloadSchemas

> Task :mlDeleteUserSchemas
Deleting user schemas in database 'data-hub-demo-final-SCHEMAS' via : cts:uris((), (), cts:not-query(cts:collection-query(('http://marklogic.com/xdmp/temporal/axis', 'http://marklogic.com/xdmp/temporal/collection', 'http://marklogic.com/xdmp/view')))) ! xdmp:document-delete(.)

[output.txt](https://github.com/marklogic/marklogic-data-hub/files/2708975/output.txt)

#### Tech details

Mac OS
MarkLogic 9.0-7.2
DHF 4.1.0
the DHF was bootstrapped with quick-start-4.1.0.war

#### The devil is in the details...

Please find attach the command output.

It seems that the tasks 'mlDeleteUserSchemas' called by 'mlReloadSchemas' deletes the final DB content instead of the final schema DB.

**Comment by aebadirad** [comment]
@rjrudin has some improvements in ml-gradle that will help this, so pinging him here on this for awareness

**Comment by rjrudin** [comment]
The task that deletes data from a schemas database is DeleteUserSchemasTask. It has this code in it:

```
String xquery
String database

@TaskAction
void deleteUserSchemas() {
	if (xquery == null) {
		xquery = "cts:not-query(" +
			"cts:collection-query((" +
			"'http://marklogic.com/xdmp/temporal/axis', " +
			"'http://marklogic.com/xdmp/temporal/collection', 'http://marklogic.com/xdmp/view'" +
			"))" +
			")"
	}

	if (database == null) {
		database = getAppConfig().getSchemasDatabaseName()
	}
```

Thus, it defaults to the value of mlAppConfig.schemasDatabaseName. That property is set in DHF in HubConfigImpl:

```
config.setSchemasDatabaseName(finalSchemasDbName);
```

Thus, the only way that I can see the final content database being cleared is if finalSchemasDbName is set to the name of the final content database. 

Note that you can always configure the task in build.gradle to specify the database that you want cleared:

```
mlDeleteUserSchemas.database = "my-schemas-database"
```

**Comment by akshaysonvane** [comment]
Hey @rjrudin, though it gets the database name from getAppConfig().getSchemasDatabaseName(), I don't see it being used in the query.
This is the output that I get
`Deleting user schemas in database 'data-hub-final-SCHEMAS' via : cts:uris((), (), cts:not-query(cts:collection-query(('http://marklogic.com/xdmp/temporal/axis', 'http://marklogic.com/xdmp/temporal/collection', 'http://marklogic.com/xdmp/view')))) ! xdmp:document-delete(.)`

I have:
mlFinalSchemasDbName=data-hub-final-SCHEMAS
mlFinalAppserverName=data-hub-FINAL

**Comment by aebadirad** [comment]
This is fixed in 4.1.1

**Comment by aebadirad** [comment]
Oops, preempted this. Will be fixed in dev shortly with the 3.11.1 update to ml-gradle plus sprints current work

**Comment by aebadirad** [comment]
Fixed in version 4.2.0
